src/internal/widgets/index.tsx

```diff
@@ -7,14 +7,19 @@ import { getGlobalFlag } from '@cloudscape-design/component-toolkit/internal';
 import { useVisualRefresh } from '../hooks/use-visual-mode';
 
 // Built-in React.FunctionComponent has always present `children` property which is not desired
-type FunctionComponent<Props> = (props: Props) => JSX.Element;
+export type FunctionComponent<Props> = (props: Props) => JSX.Element;
+type PropsType<Component extends FunctionComponent<any>> =
+  Component extends FunctionComponent<infer Props> ? Props : never;
 
-export function createWidgetizedComponent<Component extends FunctionComponent<any>>(Implementation: Component) {
+export function createWidgetizedComponent<Component extends FunctionComponent<any>>(
+  Implementation: Component,
+  Skeleton?: React.ForwardRefExoticComponent<PropsType<Component> & React.RefAttributes<HTMLElement>>
```

Comment:
New feature here, in addition to the fully working implementation we also receive a skeleton state to render while loading.